src/main/java/com/mgmtp/radio/service/youtube/YouTubeServiceImpl.java

```diff
@@ -0,0 +1,83 @@
+package com.mgmtp.radio.service.youtube;
+
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.DateTime;
+import com.google.api.services.youtube.YouTube;
+import com.google.api.services.youtube.model.SearchListResponse;
+import com.google.api.services.youtube.model.SearchResult;
+import com.mgmtp.radio.domain.youtube.YouTubeVideo;
+import com.mgmtp.radio.dto.youtube.YouTubeSearchCriteriaDTO;
+import com.mgmtp.radio.dto.youtube.YouTubeVideoDTO;
+import com.mgmtp.radio.respository.youtube.YouTubeRepository;
+import org.springframework.stereotype.Service;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+
+//@Service
+public class YouTubeServiceImpl implements YouTubeService {
+    private static final String YOUTUBE_URL = "https://www.youtube.com/watch?v=";
+    private static final long MAX_SEARCH_RESULTS = 5;
+    private final YouTubeRepository youTubeRepository;
+
+    public YouTubeServiceImpl(YouTubeRepository youTubeRepository) {
+        this.youTubeRepository = youTubeRepository;
+    }
+
+    @Override
+    public List<YouTubeVideo> fetchVideosByQuery(String queryTerm) {
+        List<YouTubeVideo> videos = new ArrayList<YouTubeVideo>();
+
+        try {
+            YouTube youTube = getYouTube();
+
+            YouTube.Search.List search = youTube.search().list("id,snippet");
+
+            String apiKey = "AIzaSyD_HCz-IjU056WTFjBgWYmjjg1YnwRPXXM";
```

Comment:
don't hard code apiKey, should put it in application.properties

Reply:
This is a redundant file. I removed it.